modules/server/src/main/scala/seqexec/server/EpicsUtil.scala

```diff
@@ -43,18 +43,21 @@ abstract class EpicsCommandBase[F[_]: Async](sysName: String) extends EpicsComma
 
   override def post(timeout: FiniteDuration): F[ApplyCommandResult] = setTimeout(timeout) *>
     Async[F]
-      .async_[ApplyCommandResult] { (f: Either[Throwable, ApplyCommandResult] => Unit) =>
-        cs.map { ccs =>
-          ccs.postCallback {
-            new CaCommandListener {
-              override def onSuccess(): Unit = f(ApplyCommandResult.Completed.asRight)
-              override def onPause(): Unit   = f(ApplyCommandResult.Paused.asRight)
-              override def onFailure(cause: Exception): Unit = f(cause.asLeft)
+      .async[ApplyCommandResult] { (f: Either[Throwable, ApplyCommandResult] => Unit) =>
+        Async[F].delay {
+          cs.map { ccs =>
+            ccs.postCallback {
+              new CaCommandListener {
+                override def onSuccess(): Unit = f(ApplyCommandResult.Completed.asRight)
+                override def onPause(): Unit   = f(ApplyCommandResult.Paused.asRight)
+                override def onFailure(cause: Exception): Unit = f(cause.asLeft)
+              }
             }
-          }
-          // It should call f on all execution paths, thanks @tpolecat
-        }.void
-          .getOrElse(f(SeqexecFailure.Unexpected("Unable to trigger command.").asLeft))
+            // It should call f on all execution paths, thanks @tpolecat
+          }.void
+            .getOrElse(f(SeqexecFailure.Unexpected("Unable to trigger command.").asLeft))
+          Some(Async[F].unit)
```

Comment:
Wouldn't this simple discard all the code above? I think the correct expression would be `Async[F].delay{...}.as(Some(Async[F].unit))`

Reply:
Oh wait, my bad. I got confused by all the brackets.
But I still think that my suggestion is  prettier :)